missilewars-plugin/src/main/java/de/butzlabben/missilewars/util/geometry/GameArea.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * This file is part of MissileWars (https://github.com/Butzlabben/missilewars).
+ * Copyright (c) 2018-2021 Daniel Nägele.
+ *
+ * MissileWars is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MissileWars is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MissileWars.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package de.butzlabben.missilewars.util.geometry;
+
+import de.butzlabben.missilewars.configuration.arena.AreaConfiguration;
+import lombok.Getter;
+import lombok.Setter;
+import org.bukkit.Location;
+import org.bukkit.World;
+
+@Getter
+public class GameArea {
+
+    private World world;
+    @Setter private Location position1, position2;
+
+    private int minX, minY, minZ;
+    private int maxX, maxY, maxZ;
+
+    public GameArea(Location pos1, Location pos2) {
+        
+        if (!Geometry.bothLocInSameWorld(pos1, pos2)) throw new IllegalArgumentException("Defined positions are not in the same world!");
+        if (pos1 == pos2) throw new IllegalArgumentException("The selected positions do not differ.");
+        
+        this.world = pos1.getWorld();
+        
+        this.position1 = pos1;
+        this.position2 = pos2;
+        
+        calculate();
+    }
+    
+    public GameArea(Location center, int offset, int high) {
```

Comment:
Better use `height` instead of `high`. Please also add Java Docs, so people know how it works. I for example assumed, that `center` would be the real center, so that
```java
        long y1 = center.getBlockY() - heigth;
        long y2 = center.getBlockY() + heigth;
```
The real behaviour as to be clear from either the constructor signature or documentation